### PR TITLE
Allow passing a non-default secrets provider to AutomationAPI

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #5297

### CODE REVIEW

File: sdk/go/x/auto/local_workspace.go [hunk 1]
```diff
@@ -136,7 +137,7 @@ func (l *LocalWorkspace) GetConfig(ctx context.Context, fqsn string, key string)
 	if err != nil {
 		return val, errors.Wrapf(err, "could not get config, unable to select stack %s", fqsn)
 	}
-	stdout, stderr, errCode, err := l.runPulumiCmdSync(ctx, "config", "get", key, "--show-secrets", "--json")
+	stdout, stderr, errCode, err := l.runPulumiCmdSync(ctx, "config", "get", key, "--json")
```
Comment (EvanBoyle) [inline sdk/go/x/auto/local_workspace.go:140]: `--show-secrets` was an intentional decision here (and below). The return type of `GetConfig` is a `ConfigValue` which has:

- `Value`: the plaintext context of the config
- `Secret`: a boolean flag indicating secretness

This allows users (who are presumably privileged) to handle secrets with the maximum flexibility possible without overcomplicating the API surface.

**Review by EvanBoyle — COMMENTED** [review]
This is looking great. Other than the comment, it would be nice to add some godoc examples:

Something similar to this, but that demonstrates creating a workspace and a stack with a secrets provider: https://github.com/pulumi/pulumi/blob/46da25492e38607967a83c0fa4bfa3f9772cb409/sdk/go/x/auto/example_test.go#L188

I believe that you can get an example entry if you do something like `ExampleLocalWorkspace_secretsProvider`:

https://godoc.org/github.com/fluhus/godoc-tricks#Examples

A test would be nice as well if possible.

File: sdk/go/x/auto/local_workspace.go (hunk 1 above)
Comment (stack72) [reply sdk/go/x/auto/local_workspace.go:140]: There is no such flag on config get - it *always* returns the unencrypted value by default

Comment (EvanBoyle) [reply sdk/go/x/auto/local_workspace.go:140]: LOL good catch :)

File: sdk/go/x/auto/local_workspace_test.go [hunk 2]
```diff
@@ -32,6 +33,71 @@ import (
...
 
+func TestWorkspaceSecretsProvider(t *testing.T) {
+	ctx := context.Background()
+	pName := "go_secrets_provider"
+	sName := fmt.Sprintf("int_test%d", rangeIn(10000000, 99999999))
+	fqsn := FullyQualifiedStackName(pulumiOrg, pName, sName)
+
+	// We can't use Workspace EnvVars as the Workspace uses the secrets provider to
+	// create the Stack
+	os.Setenv("PULUMI_CONFIG_PASSPHRASE", "password")
```
Comment (EvanBoyle) [inline sdk/go/x/auto/local_workspace_test.go:44]: Note that you can do something along the lines of: 

```go
w, err := NewLocalWorkspace( Program(...), SecretsProvider(...), Project(...))
w.SetEnv()
NewStack(ctx, fqsn, ...)
```

This is fine for now. CC @metral what do you think about adding environment variables as workspace options to make this a little more ergonomic? That way both SecretsProvider and Envs could be specified at workspace creation time and enable using the `NewStackInlineSource` helper?

**Review by EvanBoyle — APPROVED** [review]
Looks good. Just a heads up you can use LocalWorkspace directly to avoid having to set OS envs, but this is fine for now as this is just for testing.

**Comment by stack72** [comment]
> Looks good. Just a heads up you can use LocalWorkspace directly to avoid having to set OS envs, but this is fine for now as this is just for testing.

yeah my testing used exactly that - i was using this for an automated test only

File: sdk/go/x/auto/local_workspace_test.go (hunk 2 above)
Comment (metral) [reply sdk/go/x/auto/local_workspace_test.go:44]: > what do you think about adding environment variables as workspace options to make this a little more ergonomic? That way both SecretsProvider and Envs could be specified at workspace creation time

SGTM. Feel free to open an issue and assign it to me

**Pull request merged**